src/core/public/application/types.ts

```diff
@@ -625,7 +626,7 @@ export interface AppLeaveActionFactory {
    * @param text The text to display in the confirmation message
    * @param title (optional) title to display in the confirmation message
    */
-  confirm(text: string, title?: string): AppLeaveConfirmAction;
+  confirm(text: string, title?: string, callback?: () => void): AppLeaveConfirmAction;
```

Comment:
we should update the comment of this method as well.

Reply:
When the user clicks on the home button, the appId will first change, and then the page will start refreshing, resulting in two modals popping up. This callback function is designed to prevent the popping of the second modal, allowing users to smoothly navigate to the page.

Reply:
https://github.com/user-attachments/assets/3f7cc4db-bdc9-4c64-99b1-4b7c54800c35

Reply:
the comment has been updated